dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -114,8 +104,68 @@ case class BinaryOutcomeDLCWithSelf(
   val sigPubKeyLose: ECPublicKey =
     Schnorr.computePubKey(messageLose, preCommittedR, oraclePubKey)
 
+  val fundingLocalPrivKey: ECPrivateKey =
+    localExtPrivKey.deriveChildPrivKey(UInt32(0)).key
+
+  val fundingRemotePrivKey: ECPrivateKey =
+    remoteExtPrivKey.deriveChildPrivKey(UInt32(0)).key
+
+  val finalLocalPrivKey: ECPrivateKey =
+    localExtPrivKey.deriveChildPrivKey(UInt32(2)).key
+
+  val finalRemotePrivKey: ECPrivateKey =
+    remoteExtPrivKey.deriveChildPrivKey(UInt32(2)).key
+
+  private val (winIndex, loseIndex) =
+    if (outcomeWin.compareTo(outcomeLose) > 0) {
+      (1, 2)
+    } else {
+      (2, 1)
+    }
+
+  val cetLocalRefundPrivKey: ECPrivateKey = localExtPrivKey
+    .deriveChildPrivKey(
+      BIP32Path(BIP32Node(1, hardened = false), BIP32Node(0, hardened = false)))
+    .key
+
+  val cetLocalWinPrivKey: ECPrivateKey = localExtPrivKey
+    .deriveChildPrivKey(
+      BIP32Path(BIP32Node(1, hardened = false),
```

Comment:
maybe move this path construction into a helper method? The only thing that is different is the `extKey` and the `index` right?